akka-http/src/main/scala/akka/http/scaladsl/server/Route.scala

```diff
@@ -65,9 +73,29 @@ object Route {
                                 exceptionHandler: ExceptionHandler         = null): Flow[HttpRequest, HttpResponse, NotUsed] =
     Flow[HttpRequest].mapAsync(1)(asyncHandler(route))
 
+  def asyncHandler(route: Route, routingSettings: RoutingSettings, parserSettings: ParserSettings)(implicit system: ActorSystem): HttpRequest => Future[HttpResponse] = {
+    val routingLog = RoutingLog(system.log)
+    val executionContext = system.dispatcher
+    val materializer = SystemMaterializer(system).materializer
```

Comment:
One thing I thought about is that this may now be a different serializer than the http flow is materialized with, wouldn't it make more sense to use that? (The `fromMaterializer(materializer => ...)` operator, or is that only in 2.6?)